src/runtime/opencl/opencl_module.cc

```diff
@@ -108,7 +109,27 @@ class OpenCLWrappedFunc {
   LaunchParamConfig launch_param_config_;
 };
 
-OpenCLModuleNode::~OpenCLModuleNode() {
+class OpenCLSPIRVModuleNode : public OpenCLModuleNodeBase {
```

Comment:
Probably all runtime part related to SPIRV can be moved in a new directory in `src/runtime`? I suppose that SPIRV can be useful no only for Vulcan, so SPIRV can be separated from other runtimes initially. What do you think?

Reply:
This class is still specific to OpenCL. Other than the input binary format and kernel initialization, most runtime code are shared between the normal OpenCL runtime and the new one that consumes SPIRV. So I want to put it here.

Reply:
Probably in this case, `SPIRVShader` from Vulcan can be moved to OpenCL directory? And I'm ok if it is in the same files with OpenCL runtime. Just as an idea, probably the SPIRV part can be moved to new files such as `spirv_module.h/cc`?

Reply:
Ok, made the following changes:
* Add `runtime/spirv` dir and put `SPIRVShader` there
* Add `runtime/opencl_module_spirv.cc` and put `OpenCLSPIRVModuleNode` and related code there.

Overall I also like this organization. Please take a look.